custom_components/elasticsearch/const.py

```diff
@@ -34,7 +34,7 @@
 ONE_MINUTE = 60
 ONE_HOUR = 60 * 60
 
-VERSION_SUFFIX = "-v4_2"
+VERSION_SUFFIX = "-v4_3"
```

Comment:
index_mapping.json hasn't changed, can you help me with why this is needed?

Reply:
Hmm, I'm now second-guessing my memory. I found that bumping the `VERSION_SUFFIX` fixed my ingest issue, which I believed to be caused by the introduction of `valueas`. But you're right, the `index_mapping.json` hasn't changed. So, I don't see how this could have fixed what I was seeing, if my analysis of the root cause was correct. Let me revisit and try to reproduce.

Reply:
I found the problem. The config flow isn't honoring an explicit request to use the legacy index mode. I'll add a test illustrating the issue with a fix, and revert the `VERSION_SUFFIX` change.

Reply:
Reverted `VERSION_SUFFIX` in https://github.com/strawgate/homeassistant-elasticsearch/pull/1/commits/9b9d27bd0ddfdae03597d09ad7e2c7c1c210a5f7.

Fixed config flow in https://github.com/strawgate/homeassistant-elasticsearch/pull/1/commits/7b051da206c1e90d53db01758ccab7d3d19a8cc1